pkg/cmap/cmap_test.go

```
@@ -0,0 +1,202 @@
+		for i := 0; i < numItems; i++ {
+			m.Set(i, i)
+		}
+
+		var wg sync.WaitGroup
+		wg.Add(3) // For Keys, Values, and modifier goroutines
+
+		// Start a goroutine to continuously modify the map.
+		go func() {
+			defer wg.Done()
+			for i := 0; i < numItems; i++ {
+				m.Set(rand.Intn(numItems), rand.Int())
```

golangci-lint

[high] G404: Use of weak random number generator (math/rand or math/rand/v2 instead of crypto/rand)

(gosec)